lang/cs/Org.Apache.REEF.Client/YARN/IJobResourceUploader.cs

```diff
@@ -28,6 +29,6 @@ public interface IJobResourceUploader
         /// <param name="driverLocalFolderPath">Local folder where REEF application resources are staged</param>
         /// <param name="jobSubmissionDirectory">Remote directory path where we will upload resources</param>
         /// <returns>Path, modification time and size of uploaded file as JobResource</returns>
-        JobResource UploadJobResource(string driverLocalFolderPath, string jobSubmissionDirectory);
+        ICollection<JobResource> UploadJobResource(string driverLocalFolderPath, string jobSubmissionDirectory);
```

Comment:
Consider splitting this method into two: One for the archive another for the file instead of doing the two things in the implementation only. 
The benefit of doing that is that it makes what is happening behind the scenes more explicit in the interface. Further if anyone wants to provide custom the implementation for IJobResourceUploader they will explicitly know what needs to be done here. Any custom implementation should not need REEFFileNames in this layer.